### DISCUSSION TITLE
Help adding new feature: any sensor could be used

### DISCUSSION THREAD

**Discussion by raqueltac** [cyberman54/ESP32-Paxcounter#803] [Q&A]
Hello, 

I'm developing a new feature for Paxcounter using Cayenne LPP protocol and I need some help for **integrating my portion of code** in the global code. 

The idea is to add any sensor to a ESP32/ESP8266 and to send the payload in Cayenne LPP format to the ESP32 with Paxcounter code. It would make it easy to reuse code for any sensor (temperature, humidity, gyrometer, digital input, ...) without modifying Paxcounter code, just by adding 3 new files. These new files receive and process the payload sent by the tx in Cayenne LPP format. If a temperature data is received, it detects the prefix, stores the payload and does some operations (maximum and minimum value and arithmetic media). So the idea is **to send this new payload** (maximum and minimum value and arithmetic media of the temperature, in this example) **to TTN using the SENDCYCLE value**. But I'm not sure of **_how to add new data to the payload to be sent_**, could you help me?

Here is the complete code, which is working separatly from the ESP32-Paxcounter global code:
https://github.com/raqueltac/Paxcounter-Cayenne/blob/master/src/main.cpp

Thank you,
Raquel

**Comment by cyberman54** [comment]
@raqueltac The given link is not working (404)... ?

**Reply by raqueltac** [reply]
now it's working, thanks!
